core/src/main/java/com/linecorp/armeria/server/annotation/JacksonRequestConverterFunction.java

```diff
@@ -74,6 +74,11 @@ public Object convertRequest(ServiceRequestContext ctx, AggregatedHttpMessage re
                 try {
                     return reader.readValue(content);
                 } catch (JsonProcessingException e) {
+                    if (expectedResultType == byte[].class ||
+                        expectedResultType == HttpData.class) {
```

Comment:
Oh, I agree with [trustin's comment](https://github.com/line/armeria/pull/1745#discussion_r278795682).
Do we need to handle `String` too?

Reply:
Ah, I mean that we need to handle a content like `{"a": 1}`. It's not a JSON string, it's an object. So it would be failed to convert the content into a string in this converter. Also, it might be handled by the next `StringRequestConverterFunction` as converting it into a raw string `{"a": 1}`.

For the case that @trustin explained:
If the content is a JSON string like `"value"` and the `expectedResultType` is `String.class`, the conversion would be succeeded. So I think we don't need to consider the case from here.

Does it make sense? Please feel free to leave any question. :-)